ballerina/variable_validator.bal

```diff
@@ -51,7 +51,7 @@ class VariableValidator {
             self.errors.push(errorDetail);
         }
         foreach parser:Selection selection in operationNode.getSelections() {
-            self.visitSelection(selection, variableDefinitions);
+            self.visitSelection(selection, <anydata>variableDefinitions);
```

Comment:
Why do we need to cast here?

Reply:
since `variableDefinition` is `map<parser:VariableDefinition>` type, It will return a compilation error without cast.

Reply:
How's that possible? `map<parser:VariableDefinition>` is a map of record type, which does not have any object references, right? It should be `anydata` by default. Shall we create an issue for this in lang repo?

Reply:
may be due to this `ArgumentValue|ArgumentNode defaultValue?;` field in `VariableDefinition`. `ArgumentNode` is an object type.

Reply:
In that case, you cannot cast this to `anydata`. This will panic when there's an `ArgumentNode` value.